servers/physics_3d/soft_body_3d_sw.cpp

```diff
@@ -899,26 +902,64 @@ void SoftBody3DSW::add_velocity(const Vector3 &p_velocity) {
 }
 
 void SoftBody3DSW::apply_forces() {
-	if (pressure_coefficient < CMP_EPSILON) {
+	int ac = areas.size();
+	bool stopped = false;
+
+	if (pressure_coefficient < CMP_EPSILON && ac <= 0) {
 		return;
 	}
 
 	if (nodes.is_empty()) {
 		return;
 	}
 
-	uint32_t i, ni;
+	if (ac) {
+		areas.sort();
+	}
+
+	uint32_t i, ni, j;
 
-	// Calculate volume.
 	real_t volume = 0.0;
 	const Vector3 &org = nodes[0].x;
+
+	// Iterate over faces (try not to iterate elsewhere if possible)
 	for (i = 0, ni = faces.size(); i < ni; ++i) {
 		const Face &face = faces[i];
+
+		Vector3 wind_force(0, 0, 0);
+
+		// Compute volume
 		volume += vec3_dot(face.n[0]->x - org, vec3_cross(face.n[1]->x - org, face.n[2]->x - org));
+
+		// Compute nodal forces from area winds
+		if (ac) {
+			const AreaCMP *aa = &areas[0];
+			for (int i = ac - 1; i >= 0; i--) {
```

Comment:
It's missing exiting the loop when stopped is true, and setting stopped to true on REPLACE mode like gravity does. Otherwise it doesn't respect the replace override mode as it keeps going through more areas.